src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -685,6 +698,14 @@ private bool EnableDelisting
                 return _getSetting("enableDelisting", false);
             }
         }
+        private int RetentionRule
+        {
+            get
+            {
+                // If the setting is misconfigured, treat it as off (backwards compatibility).
+                return GetIntAppSetting("retentionRule", 0);
```

Comment:
Yes, I activate this rule only if the value is strictly higher to 0 (if (PackageVersionRetention > 0)).